pkg/cloud/api/check.go

```diff
@@ -171,3 +171,42 @@ func checkSchema(t reflect.Type) error {
 
 	return nil
 }
+
+// CheckConversion checks if we can convert type From to type To. Conversion is
+// possible if type From has the subset of fields with the same type and name as
+// in To type. This function is not symmetric. Path parameter is used for better
+// error reporting.
+func CheckConversion(p Path, from, to reflect.Type) error {
```

Comment:
renamed to `CheckStructuralSubset` because as Hugo noticed isXX should return bool and checkXX error. I want to return error here so we have the information which field is missing in subset.